jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/profiles/ProfileCredentialProviderFactory.kt

```diff
@@ -114,6 +114,21 @@ class ProfileCredentialProviderFactory(private val ssoCache: SsoCache = diskCach
             }
         }
 
+        // any profiles with a modified 'source_profile' need to be marked as well
+        newProfiles.validProfiles.forEach { (_, profile) ->
+            val profileId = profile.asId(newProfiles.validProfiles)
+            if (profileId in profilesModified) {
+                // already marked; skip
+                return@forEach
+            }
+            for (source in profile.traverseCredentialChain(newProfiles.validProfiles)) {
```

Comment:
Do we need to reverse the chain?

If you have `profile 1 (root`) -> `profile 2` -> `profile 3 (leaf)`, and `profile 1` is modified, then when we traverse 3->2->1...2 will not be in the profilesModified list yet...leading to only profile 1 and 2 being notified.

Reply:
Nvm. This code ends up being logically equivalent to `profile.traverseCredentialChain(newProfiles.validProfiles).any {it in profilesModified }`

Reply:
Can we at least add the above scenario to a unit test? Or just extend `testModifyingASourceProfile` to use 3 profiles instead of 2.

Reply:
added additional tests